Qualifiers/src/androidTest/java/com/microsoft/device/display/samples/qualifiers/QualifiersTest.kt

```diff
@@ -6,87 +6,63 @@
-        onView(withId(R.id.text))
-            .check(matches(withText("Hello World! \nSingle screen mode")))
-        onView(withId(R.id.resource_folder_text))
-            .check(matches(withText("sw540dp-1350x1800")))
+    @After
+    fun tearDown() {
+        unfreezeRotation()
     }
 
     @Test
-    fun displayDualLayout_inDualMode() {
-        spanApplication()
+    fun singleScreenModeInPortrait() {
+        checkContent(null)
+    }
 
-        onView(withId(R.id.layout_container))
-            .check(matches(withBackgroundColor(R.color.colorAccent)))
-        onView(withId(R.id.image))
-            .check(matches(withContentDescription("Dual screen image")))
-        onView(withId(R.id.text))
-            .check(matches(withText("Hello World! \nSpanned mode!")))
-        onView(withId(R.id.resource_folder_text))
-            .check(matches(withText("sw720dp-2784x1800")))
+    @Test
+    fun dualScreenModeInPortrait() {
+        switchFromSingleToDualScreen()
+        checkContent(R.color.colorAccent)
     }
 
-    private fun spanApplication() {
-        device.swipe(675, 1780, 1350, 900, 400)
+    @Test
+    fun singleScreenModeInLandscape() {
+        setOrientationLeft()
+        checkContent(null)
     }
 
-    companion object {
-        fun withBackgroundColor(@ColorInt expectedColor: Int?): Matcher<View> =
-            object : BoundedMatcher<View, ConstraintLayout>(ConstraintLayout::class.java) {
-                override fun describeTo(description: Description?) {
-                    description?.appendText(
-                        "Check for item to have color with id $expectedColor"
-                    )
-                }
+    @Test
+    fun dualScreenModeInLandscape() {
+        switchFromSingleToDualScreen()
+        setOrientationLeft()
+        checkContent(R.color.colorAccent)
+    }
 
-                override fun matchesSafely(view: ConstraintLayout?): Boolean {
-                    view?.let {
-                        if (expectedColor == null) {
-                            return view.background == null
-                        }
-                        if (view.background is ColorDrawable) {
-                            val actualColor = (view.background as ColorDrawable).color
-                            return actualColor == view.context.getColor(expectedColor)
-                        }
-                    }
-                    return false
-                }
-            }
+    private fun checkContent(@ColorInt backgroundColor: Int?) {
+        onView(withId(R.id.layout_container)).check(matches(hasBackgroundColor(backgroundColor)))
+        onView(withId(R.id.image)).check(matches(withContentDescription(R.string.sticker_image_content_description)))
+        onView(withId(R.id.message)).check(matches(withText(R.string.hello_world)))
+        onView(withId(R.id.resource_folder)).check(matches(withText(R.string.resource_folder)))
```

Comment:
The previous tests were using strings instead of string resources in order to be sure that the qualifiers are set correctly. For example if the additional qualifier layouts are removed, the last three tests will still pass. Now the verification is only based on the color change, wouldn't it be better to also test that the texts are the right ones?

Reply:
Done. I didn't think about this possibility. Now, I'm checking directly the strings.

Reply:
Cool, no worries! 😃